internal/serve/httphandler/payments_handler.go

```diff
@@ -177,6 +165,29 @@ func (p PaymentsHandler) RetryPayments(rw http.ResponseWriter, req *http.Request
 	httpjson.RenderStatus(rw, http.StatusOK, map[string]string{"message": "Payments retried successfully"}, httpjson.JSON)
 }
 
+func (p PaymentsHandler) producePaymentsReadyEvents(ctx context.Context, payments []*data.Payment) error {
+	msg, msgErr := events.NewMessage(ctx, events.PaymentReadyToPayTopic, "", events.PaymentReadyToPayRetryFailedPayment, nil)
+	if msgErr != nil {
+		return fmt.Errorf("creating a new message: %w", msgErr)
+	}
+
+	paymentsReadyToPay := schemas.EventPaymentsReadyToPayData{TenantID: msg.TenantID}
+	for _, payment := range payments {
+		paymentsReadyToPay.Payments = append(paymentsReadyToPay.Payments, schemas.PaymentReadyToPay{ID: payment.ID})
+	}
+	msg.Data = paymentsReadyToPay
+
+	if p.EventProducer != nil {
+		err := p.EventProducer.WriteMessages(ctx, *msg)
+		if err != nil {
+			return fmt.Errorf("writing message %s on event producer: %w", msg, err)
+		}
+	} else {
+		log.Ctx(ctx).Errorf("event producer is nil, could not publish message %s", msg.String())
```

Comment:
Assuming there's no sensitive data in the message, It'd be beneficial to use a verb that provides the maximum output, so we can debug things more easily:
```suggestion
		log.Ctx(ctx).Errorf("event producer is nil, could not publish message %+v", msg)
```

Reply:
We're just passing IDs. Fixed here and in other locations.